pkg/comp-functions/functions/vshnpostgres/postgresql_deploy.go

```diff
@@ -671,3 +687,31 @@ func createCopyJob(comp *vshnv1.VSHNPostgreSQL, svc *runtime.ServiceRuntime) err
 
 	return nil
 }
+
+func setClusterUnreadyIfProfilesNotEqual(svc *runtime.ServiceRuntime) {
+	desProfile := &sgv1.SGInstanceProfile{}
+	obsProfile := &sgv1.SGInstanceProfile{}
+
+	changed := false
+
+	err := svc.GetObservedKubeObject(obsProfile, "profile")
```

Comment:
Isn't this the other way around? on first cycle the desired is missing and not the observed?

Reply:
You mixed it up :) The desired state gets set during the comp function in `createSgInstanceProfile(ctx, comp, svc)` and the observed state will only be available after the first cycle, sometimes even after a few, depending on how fast provider-kubernetes is.

Reply:
ah I see.. chicken egg problem :D who's first?